driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -451,7 +481,7 @@ private Connection findLeastBusyForShard(int shardId) {
             >= Math.min(
                 leastBusy.maxAvailableStreams(),
                 options().getMaxRequestsPerConnection(hostDistance))) {
-          return enqueue(timeout, unit, maxQueueSize, shardId);
```

Comment:
I think this is wrong. If we can't find connection we should enqueue for the requested shardId